code/game/objects/items/tools/tool_behaviour.dm

```diff
@@ -39,8 +39,8 @@
 
 // Called before use_tool if there is a delay, or by use_tool if there isn't.
 // Only ever used by welding tools and stacks, so it's not added on any other use_tool checks.
-/obj/item/proc/tool_start_check(mob/living/user, amount=0)
-	return tool_use_check(user, amount)
+/obj/item/proc/tool_start_check(atom/target, mob/living/user, amount=0)
```

Comment:
You don't need to pass the target to this proc, this is used only to check if the tool can be used (which is independent of what the target is)